packages/compass-e2e-tests/tests/collection-indexes-tab.test.ts

```diff
@@ -88,4 +95,73 @@ describe('Collection indexes tab', function () {
 
     await indexComponent.waitForDisplayed({ reverse: true });
   });
+
+  describe('server version 6.1.0 with env variable COMPASS_COLUMNSTORE_INDEXES = true', function () {
+    let initialEnvVars: NodeJS.ProcessEnv;
+
+    before(function () {
+      initialEnvVars = Object.assign({}, process.env);
+
+      process.env.COMPASS_COLUMNSTORE_INDEXES = 'true';
+    });
+
+    after(function () {
+      process.env = initialEnvVars;
+    });
+
+    it('supports creating a columnstore index', async function () {
+      if (semver.lt(MONGODB_VERSION, '6.1.0')) {
+        return this.skip();
+      }
+
+      await browser.clickVisible(Selectors.CreateIndexButton);
+
+      const createModal = await browser.$(Selectors.CreateIndexModal);
+      await createModal.waitForDisplayed();
+
+      await browser.clickVisible(Selectors.CreateIndexModalFieldSelect);
+
+      const fieldList = await browser.$(
+        `${Selectors.CreateIndexModalFieldSelect} [role="listbox"]`
+      );
+      await fieldList.waitForDisplayed();
+
+      const columnstoreIndexKey = '$**';
+      await fieldList.setValue(columnstoreIndexKey);
+
+      await browser.clickVisible(Selectors.CreateIndexModalTypeSelect);
+      const typeList = await browser.$(
+        `${Selectors.CreateIndexModalTypeSelect} [role="listbox"]`
+      );
+      await typeList.waitForDisplayed();
+      // Click on the columnstore index type.
+      const textOption = await typeList.$('div=columnstore'); // div element with the text "columnstore"
+      textOption.waitForDisplayed();
+      textOption.click();
+
+      await browser.clickVisible(Selectors.CreateIndexConfirmButton);
+
+      await createModal.waitForDisplayed({ reverse: true });
+
+      const indexComponent = await browser.$(
+        Selectors.indexComponent('columnstore')
```

Comment:
@lerouxb Added a feature flag, `COMPASS_COLUMNSTORE_INDEXES`, to enable the columnstore index creation feature. Also added this e2e test. Looks like the default name for indexes created will be `columnstore` from https://github.com/mongodb/mongo/blob/master/src/mongo/db/index_names.cpp
When we remove the feature flag we'll know if the test is failing as well.

Once tests pass I'll merge tomorrow if that's cool.

Reply:
Sounds good!